xskillscore/core/np_deterministic.py

```diff
@@ -219,3 +284,107 @@ def _mae(a, b, weights, axis):
         )
     else:
         return absolute_error.mean(axis=axis)
+
+
+def _mad(a, b, weights, axis):
+    """
+    Median Absolute Error.
+
+    Parameters
+    ----------
+    a : ndarray
+        Input array.
+    b : ndarray
+        Input array.
+    axis : int
+        The axis to apply the mae along.
+
+    Returns
+    -------
+    res : ndarray
+        Median Absolute Error.
+
+    See Also
+    --------
+    sklearn.metrics.median_absolute_error
+
+    """
+    weights = _check_weights(weights)
+
+    absolute_error = np.absolute(a - b)
+    return absolute_error.median(axis=axis)
+
+
+def _mape(a, b, weights, axis):
+    """
+    Mean Absolute Percentage Error.
+
+    Parameters
+    ----------
+    a : ndarray
+        Input array (truth to be divided by).
+    b : ndarray
+        Input array.
+    axis : int
+        The axis to apply the mae along.
+    weights : ndarray
+        Input array.
+
+    Returns
+    -------
+    res : ndarray
+        Mean Absolute Percentage Error.
+
+    See Also
+    --------
+    sklearn.metrics.mean_absolute_error / bo.absolute(a) * 100
+
+    """
+    weights = _check_weights(weights)
+    # check whether a as zeros ?
```

Comment:
Perhaps in high-level function with `xarray` funcs? How is this handled in sklearn?